example/dummy/dummy.go

```diff
@@ -46,12 +45,12 @@ func (app *DummyApplication) Commit() types.Result {
 
 func (app *DummyApplication) Query(reqQuery types.RequestQuery) (resQuery types.ResponseQuery) {
 	if reqQuery.Prove {
-		value, proof, exists := app.state.Proof(reqQuery.Data)
+		value, proof, err := app.state.GetWithProof(reqQuery.Data)
 		resQuery.Index = -1 // TODO make Proof return index
 		resQuery.Key = reqQuery.Data
 		resQuery.Value = value
-		resQuery.Proof = proof
-		if exists {
+		resQuery.Proof = proof.Bytes()
```

Comment:
In the else condition i.e when `err != nil` I wonder if instead the iavl library could pass back a sentinel error that we can then explicitly check e.g
`if err == iavl.ErrNotExists` because perhaps GetWithProof encounters another error for which this query should entirely abort.

Just a thought though.